service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.Paginator;
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.dto.PaginationContainer;
+import com.epam.esm.dto.UserDTO;
+import com.epam.esm.entity.User;
+import com.epam.esm.error.ExceptionCauseCode;
+import com.epam.esm.exception.ResourceNotFoundedException;
+import com.epam.esm.mapper.UserDTOMapper;
+import com.epam.esm.service.UserService;
+import com.epam.esm.validator.PaginationContainerValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserDao userDao;
+    private final UserDTOMapper userDTOMapper;
+    private final PaginationContainerValidator paginationContainerValidator;
+    private final Paginator<User> paginator;
+
+    @Autowired
+    public UserServiceImpl(UserDao userDao, UserDTOMapper userDTOMapper, PaginationContainerValidator paginationContainerValidator,
+                           Paginator<User> paginator) {
+        this.userDao = userDao;
+        this.userDTOMapper = userDTOMapper;
+        this.paginationContainerValidator = paginationContainerValidator;
+        this.paginator = paginator;
+    }
+
+    @Override
+    public List<UserDTO> finaAll(PaginationContainer paginationContainer) {
+        paginationContainerValidator.isPaginationContainerValid(paginationContainer);
+        List<User> userList = userDao.findAll();
+        List<User> paginateList = paginator.paginate(userList, paginationContainer);
```

Comment:
Paginating the results in the service layer seems to be meaningless because if you have 100000000 users, all they will be retrieved from DB, and only then part of them will be returned to the client. Then, if we try to get the next page of users than 100000000 users will be retrieved again from DB.
So you do pagination on the DB level, you should retrieve not all items but the items for the current page. 
Rework pagination for all the entities in your code

Reply:
Removed Paginator and added to sql parameter LIMIT